pkg/minikube/cluster/cluster_provisioner.go

```diff
@@ -0,0 +1,293 @@
+	dockerDir := p.GetDockerOptionsDir()
+	authOptions := p.GetAuthOptions()
+
+	// due to windows clients, we cannot use filepath.Join as the paths
+	// will be mucked on the linux hosts
+	authOptions.CaCertRemotePath = path.Join(dockerDir, "ca.pem")
+	authOptions.ServerCertRemotePath = path.Join(dockerDir, "server.pem")
+	authOptions.ServerKeyRemotePath = path.Join(dockerDir, "server-key.pem")
+
+	return authOptions
+}
+
+
+func (provisioner *MinishiftProvisioner) dockerDaemonResponding() bool {
+	log.Debug("checking docker daemon")
+
+	if out, err := provisioner.SSHCommand("sudo docker version"); err != nil {
+		log.Warnf("Error getting SSH command to check if the daemon is up: %s", err)
+		log.Debugf("'sudo docker version' output:\n%s", out)
+		return false
+	}
+
+	// The daemon is up if the command worked.  Carry on.
+	return true
+}
+
+func (provisioner *MinishiftProvisioner) Provision(swarmOptions swarm.Options, authOptions auth.Options, engineOptions engine.Options) error {
+	provisioner.SwarmOptions = swarmOptions
+	provisioner.AuthOptions = authOptions
+	provisioner.EngineOptions = engineOptions
+	swarmOptions.Env = engineOptions.Env
+
+	if err := provisioner.SetHostname(provisioner.Driver.GetMachineName()); err != nil {
+		return err
+	}
+
+	if err := provisioner.Subscription(); err != nil {
+		return  err
+	}
+
+
+	if err := mcnutils.WaitFor(provisioner.dockerDaemonResponding); err != nil {
+		return err
+	}
+
+	if err := makeDockerOptionsDir(provisioner); err != nil {
+		return err
+	}
+
+	provisioner.AuthOptions = setRemoteAuthOptions(provisioner)
+
+	if err := provision.ConfigureAuth(provisioner); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (provisioner *MinishiftProvisioner) GenerateDockerOptions(dockerPort int) (*provision.DockerOptions, error) {
+	var (
+		engineCfg  bytes.Buffer
+		configPath = provisioner.DaemonOptionsFile
+	)
+
+	driverNameLabel := fmt.Sprintf("provider=%s", provisioner.Driver.DriverName())
+	provisioner.EngineOptions.Labels = append(provisioner.EngineOptions.Labels, driverNameLabel)
+
+	// systemd / redhat will not load options if they are on newlines
+	// instead, it just continues with a different set of options; yeah...
+	engineConfigTemplate := ""
+	if provisioner.GetRedhatRelease(){
+		engineConfigTemplate = engineConfigTemplateRHEL
+	} else {
+		engineConfigTemplate = engineConfigTemplateCentOS
+	}
+
+	t, err := template.New("engineConfig").Parse(engineConfigTemplate)
+	if err != nil {
+		return nil, err
+	}
+
+	engineConfigContext := provision.EngineConfigContext{
+		DockerPort:       dockerPort,
+		AuthOptions:      provisioner.AuthOptions,
+		EngineOptions:    provisioner.EngineOptions,
+		DockerOptionsDir: provisioner.DockerOptionsDir,
+	}
+
+	t.Execute(&engineCfg, engineConfigContext)
+
+	daemonOptsDir := configPath
+	return &provision.DockerOptions{
+		EngineOptions:     engineCfg.String(),
+		EngineOptionsPath: daemonOptsDir,
+	}, nil
+}
```

Comment:
Whatever editor you're using (I recommend vscode btw), it should do autoformatting for you. We should also ensure we have a check in CI before PR merges (we may already have that - can't remember).